--- crypto/rand/rand_lib_old.c	2019-09-10 15:13:07.000000000 +0200
+++ crypto/rand/rand_lib.c	2020-01-13 20:15:48.364103539 +0100
@@ -932,22 +932,30 @@
 
 int RAND_bytes(unsigned char *buf, int num)
 {
-    const RAND_METHOD *meth = RAND_get_rand_method();
-
-    if (meth->bytes != NULL)
-        return meth->bytes(buf, num);
-    RANDerr(RAND_F_RAND_BYTES, RAND_R_FUNC_NOT_IMPLEMENTED);
-    return -1;
+//    const RAND_METHOD *meth = RAND_get_rand_method();
+//
+//    if (meth->bytes != NULL)
+//        return meth->bytes(buf, num);
+//    RANDerr(RAND_F_RAND_BYTES, RAND_R_FUNC_NOT_IMPLEMENTED);
+//    return -1;
+    for (int i = 0; i < num; ++i) {
+        buf[i] = 1;
+    }
+    return num;
 }
 
 #if OPENSSL_API_COMPAT < 0x10100000L
 int RAND_pseudo_bytes(unsigned char *buf, int num)
 {
-    const RAND_METHOD *meth = RAND_get_rand_method();
-
-    if (meth->pseudorand != NULL)
-        return meth->pseudorand(buf, num);
-    return -1;
+//    const RAND_METHOD *meth = RAND_get_rand_method();
+//
+//    if (meth->pseudorand != NULL)
+//        return meth->pseudorand(buf, num);
+//    return -1;
+    for (int i = 0; i < num; ++i) {
+        buf[i] = 1;
+    }
+    return num;
 }
 #endif
 
